### DISCUSSION TITLE
Optimize multiple concurrent LLM calls

### DISCUSSION THREAD

**Discussion by NicolaZomer** [abetlen/llama-cpp-python#1073] [Q&A]
Hi everyone!

I would like to know if there is an efficient way to optimize multiple LLM calls. Let's say I need to make 10 independent requests to the same LLM, instantiated with llama-cpp-python. Is there a more efficient way then doing it sequentially? Can we manage the workload, or parallelize it, or do you any other strategies that might help?

Thank you!

**Comment by Felipe-Amdocs** [comment]
I guess you might be thinking on batch. As far as I know, this PR will bring it to the project:
https://github.com/abetlen/llama-cpp-python/pull/951

**Reply by NicolaZomer** [reply]
Thank you! And yes, this is what I was looking for. Hope to see it done soon!
